### PR TITLE
Fix v1.0.4 breaking changes

### PR DESCRIPTION
**Pull request by fatmali**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +66,7 @@
   },
   "dependencies": {
     "react": "^16.9.0",
+    "fhir.js": "0.0.20",
```
Comment (internettrans) [inline package.json]: fhir.js is bundled into esm-api as a build-time module. The `dependencies` section should only be used for in-browser dependencies in the import map.

So this change is incorrect.

File: src/shared-api-objects/user-has-access-react.component.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ export default function UserHasAccessReact(props) {
 
   React.useEffect(() => {
     const subscription = getCurrentUser({ includeAuthStatus: false }).subscribe(
-      x => {
+      (x: any) => {
```
Comment (internettrans) [inline src/shared-api-objects/user-has-access-react.component.tsx:9]: What breaking change are you talking about with 1.0.4? The build passed - https://travis-ci.org/openmrs/openmrs-esm-api/builds/592146027, and it looks like openmrs-spa.org is still working, too.

File: package.json (hunk 1 above)
Comment (internettrans) [reply package.json]: Wait is this for typescript errors? That would be unfortunate.

The way that packmap works for packaging offline things is that it assumes all `dependencies` are in-browser dependencies. Could we perhaps use [`@types/fhir`](https://unpkg.com/browse/@types/fhir@0.0.30/) instead to solve this? Packmap knows to skip all the `@types` packages in the import map

Comment (fatmali) [reply package.json]: Funny thing is the build works okay by itself, but once you install it as node module there are typescript errors and `fhir.js` not found. Seems like fhir.js isn't included in the build if it's a devDependency

File: src/shared-api-objects/user-has-access-react.component.tsx (hunk 2 above)
Comment (fatmali) [reply src/shared-api-objects/user-has-access-react.component.tsx:9]: This addresses the typescript error

File: package.json (hunk 1 above)
Comment (internettrans) [reply package.json]: Yeah dependencies are the only thing installed when another package does `npm install @openmrs/esm-api`.

Can you confirm if adding `@types/fhir` as a dependency works for you? See https://www.npmjs.com/package/@types/fhir.

To test, you can use `npm link`. See https://docs.npmjs.com/cli/link.html

Comment (fatmali) [reply package.json]: works now after adding `@types/fhir`

**Review by internettrans — APPROVED** [review]

**Pull request merged**